Sources/Core/WorkingDirectory.swift

```diff
@@ -0,0 +1,12 @@
+/// This function will attempt to get the current
+/// working directory of the application
+public func workingDirectory() -> String {
+    let file = #file
+    let directory: String?
+    if file.contains(".build") {
+        directory = file.components(separatedBy: "/.build").first
+    } else {
+        directory = file.components(separatedBy: "/Sources").first
```

Comment:
We should also have one for `Packages`. This is used in 3.1 when you do `swift package edit`